components/src/index.d.ts

```diff
@@ -19,4 +19,11 @@
  *
  */
 
+import IobioCoverageDepth from './coverage_depth';
+import IobioDataBroker from './data_broker';
+import IobioHistogram from './histogram';
+import IobioPercentBox from './percent_box';
+
 declare module 'components';
```

Comment:
Correct .d.ts usage has been a point of confusion for me.
So removing `declare module` does seem to be fine.
But if I then remove the imports/exports you get the `File ... is not a module` error in App.tsx
I've used import/export syntax so types are implicit
I attempted using tsc to generate a .d.ts file but a) it caused errors and b) was essentially just a variation on this import/export syntax except this version works

Reply:
Not sure what "I've used import/export syntax so types are implicit" this part means.
I just mean `declare namespace...` isn't needed.

I think it's confusing because pnpm (npm under the hood) are doing their "best guesses" to make things work as intended. So if there's an `.d.ts` it will pick that, regardless of what's in it, even if there's nothing there. If there isn't a `.d.ts` file I think it will look at your `index.js`.

The best thing to do, because we're still in set up stage, is to finish the `build process` for the components module.
The project is still referencing the project files vs a built library.
So it's worth having the components building correctly as a pnpm module, and then using the types from the build library. It's work that has to be done at some point anyway.

Reply:
something like: 
```js
import { defineConfig } from "vite";
import { resolve } from "path";
import dts from "vite-plugin-dts";

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [dts({ rollupTypes: true })],
  build: {
    sourcemap: true,
    lib: {
      entry: resolve(__dirname, "src/index.ts"),
      name: "components",
      fileName: "index",
      formats: ["es"],
    },
    rollupOptions: {
      external: ["react", "react/jsx-runtime"],
    },
  },
});
```
for your component vite config should get you 99% of the way there (need to install vite-plugin-dts)

Reply:
Sorry -- we have discussed this and it is definitely the missing piece of the puzzle
Vite  gets the app up and running which gives the impression that everything is working as intended
But really there's a crucial step missing and we're just seeing side effects and I'm chasing down trying to fix the side effects

Reply:
> Correct .d.ts usage has been a point of confusion for me.

We basically never need to write our own .d.ts files. Declare your types and export them using the `export` keyword. TS will compile the .d.ts files for our module when the build runs.

There are times where we may use this tool but they are exceedingly rare and usually are done to "fix" the types form another module. I can show you an example if you want but its not that important.

Reply:
TS config + build updated here: [New Package TS Config](https://github.com/overture-stack/iobio-components/pull/24/commits/70e4e8cd682d6ef2238524e612e14699d1053bd8)

Reply:
https://github.com/overture-stack/iobio-components/pull/24#discussion_r1665841324